pkg/porter/parameters.go

```diff
@@ -0,0 +1,306 @@
+	}
+
+	err = p.Parameters.Validate(paramSet)
+	if err != nil {
+		return errors.Wrap(err, "parameters are invalid")
+	}
+
+	paramSet.Modified = time.Now()
+	err = p.Parameters.Save(paramSet)
+	if err != nil {
+		return errors.Wrap(err, "unable to save parameters")
+	}
+
+	return nil
+}
+
+// ShowParameter shows the parameter set corresponding to the provided name, using
+// the provided printer.PrintOptions for display.
+func (p *Porter) ShowParameter(opts ParameterShowOptions) error {
+	paramSet, err := p.Parameters.Read(opts.Name)
+	if err != nil {
+		return err
+	}
+
+	switch opts.Format {
+	case printer.FormatJson:
+		return printer.PrintJson(p.Out, paramSet)
+	case printer.FormatYaml:
+		return printer.PrintYaml(p.Out, paramSet)
+	case printer.FormatTable:
+		// Set up human friendly time formatter
+		now := time.Now()
+		tp := dtprinter.DateTimePrinter{
+			Now: func() time.Time { return now },
+		}
+
+		// Here we use an instance of olekukonko/tablewriter as our table,
+		// rather than using the printer pkg variant, as we wish to decorate
+		// the table a bit differently from the default
+		var rows [][]string
+
+		// Iterate through all ParameterStrategies and add to rows
+		for _, pset := range paramSet.Parameters {
+			sourceVal, sourceType := GetParameterSourceValueAndType(pset.Source)
+			rows = append(rows, []string{pset.Name, sourceVal, sourceType})
+		}
+
+		// Build and configure our tablewriter
+		table := tablewriter.NewWriter(p.Out)
+		table.SetCenterSeparator("")
+		table.SetColumnSeparator("")
+		table.SetAlignment(tablewriter.ALIGN_LEFT)
+		table.SetHeaderAlignment(tablewriter.ALIGN_LEFT)
+		table.SetBorders(tablewriter.Border{Left: false, Right: false, Bottom: false, Top: true})
+		table.SetAutoFormatHeaders(false)
+
+		// First, print the ParameterSet metadata
+		fmt.Fprintf(p.Out, "Name: %s\n", paramSet.Name)
+		fmt.Fprintf(p.Out, "Created: %s\n", tp.Format(paramSet.Created))
+		fmt.Fprintf(p.Out, "Modified: %s\n\n", tp.Format(paramSet.Modified))
+
+		// Now print the table
+		table.SetHeader([]string{"Name", "Local Source", "Source Type"})
+		for _, row := range rows {
+			table.Append(row)
+		}
+		table.Render()
+		return nil
+	default:
+		return fmt.Errorf("invalid format: %s", opts.Format)
+	}
+}
+
+// GetParameterSourceValueAndType takes a given valuesource.Source struct and
+// returns the source value itself as well as source type, e.g., 'path', 'env', etc.,
+// both in their string forms
+func GetParameterSourceValueAndType(pset valuesource.Source) (value string, key string) {
+	return pset.Value, pset.Key
+}
+
+// ParameterDeleteOptions represent options for Porter's parameter delete command
+type ParameterDeleteOptions struct {
+	Name string
+}
+
+// DeleteParameter deletes the parameter set corresponding to the provided
+// names.
+func (p *Porter) DeleteParameter(opts ParameterDeleteOptions) error {
+	err := p.Parameters.Delete(opts.Name)
+	if err == crud.ErrRecordDoesNotExist {
```

Comment:
I've found that with plugins the check should be this instead

```suggestion
	if err != nil && strings.Contains(err.Error(), crud.ErrRecordDoesNotExist) {
```